python/activator/middleware_interface.py

```diff
@@ -270,13 +270,20 @@ def __init__(self, central_butler: Butler, image_bucket: str, visit: FannedOutVi
         self._init_ingester()
         self._init_visit_definer()
 
-        # HACK: explicit collection gets around the fact that we don't have any
-        # timestamp/exposure information in a form we can pass to the Butler.
-        # This code will break once cameras start being versioned.
-        self.camera = self.central_butler.get(
-            "camera", instrument=self.instrument.getName(),
-            collections=self.instrument.makeUnboundedCalibrationRunName()
-        )
+        try:
+            self.camera = self.central_butler.get(
+                "camera",
+                instrument=self.instrument.getName(),
+                collections=self.instrument.makeCalibrationCollectionName(),
+            )
+        except lsst.daf.butler.DatasetNotFoundError:
+            # Repos that don't allow retrieval of camera from <instrument>/calibs
+            # have <instrument>/calibs/unbounded.
+            self.camera = self.central_butler.get(
+                "camera",
+                instrument=self.instrument.getName(),
+                collections=self.instrument.makeUnboundedCalibrationRunName(),
+            )
```

Comment:
I think this is a possible alternative:
```
camera_ref = self.central_butler.find_dataset(
    "camera",
    instrument=self.instrument.getName(),
    collections=self.instrument.makeCalibrationCollectionName(),
    timespan=Timespan(begin=None, end=None)
)
self.camera = butler.get(camera_ref)
```
it's a shame `Butler.get` doesn't accept `timespan` and forward that down to its own `find_dataset` call directly.

Your original looks fine, but if what I've written works, it'll avoid dependence on one of our shakier convention.

Reply:
I will look at forwarding timespan in [DM-43499](https://rubinobs.atlassian.net/browse/DM-43499).

Reply:
I'd like to give it a shot, but the "live" tests have turned up serious problems that will keep me busy up until the code freeze. I'll add a TODO comment and try your suggestion as part of [DM-42825](https://rubinobs.atlassian.net/issues/DM-42825), which already involves expanding our use of `find_dataset`.

Reply:
This support was added in lsst/daf_butler#982